### DISCUSSION TITLE
Regexp presets

### DISCUSSION THREAD

**Discussion by florian-lefebvre** [directus/directus#17280] [Feature Requests]
Hi, I think it would be nice to add basic regexp presets to directus (to be used in filters for example). Here are a few examples: url, email, phone number

**Comment by br41nslug** [comment]
Interesting idea, for basic formats that would be easy but things get a little more tricky for `email`s and `phone number`s as they're very hard to properly write regex for that allows all possible international options. Kind of reminds me of this article https://elliotchance.medium.com/its-impossible-to-validate-an-email-address-ae606ad711b4

**Comment by florian-lefebvre** [comment]
If this idea is adopted, I think it could done this way:

- Custom Regexp can be created in the project settings (or equivalent location) so that it's accessible in the whole project
- In locations of Directus where Regexp can be chosen, user can either choose to write a custom one or choose among the project's ones
- The project regexps would have default values like the one I was mentioning above

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
